### PR TITLE
NMS-13040: updated text for rescan options prompt in UI

### PR DESCRIPTION
**Pull request by Bonrob2**
NMS-13040: updated text for rescan options prompt in UI

### All Contributors

* [ x] Have you read and followed our [Contribution Guidelines](https://github.com/OpenNMS/opennms/blob/develop/CONTRIBUTING.md)?
* [ x] Have you [made an issue in the OpenNMS issue tracker](https://issues.opennms.org)?<br>If so, you should:
  1. update the title of this PR to be of the format: `${JIRA-ISSUE-NUMBER}: subject of pull request`
  2. update the JIRA link at the bottom of this comment to refer to the real issue number
  3. prefix your commit messages with the issue number, if possible
* [ x] Have you made a comment in that issue which points back to this PR?
* [ x] Have you updated the JIRA link at the bottom of this comment to link to your issue?
* [ x] If this is a new or updated feature, is there documentation for the new behavior?
* [x ] If this is new code, are there unit and/or integration tests?
* [ x] If this PR targets a `foundation-*` branch, does it avoid changing files in `$OPENNMS_HOME/etc/`?

### Pull Request Process

One or more reviewers should be assigned to each PR.

If you know that a particular person is subject matter expert in the area your PR affects, feel free to assign one or more reviewers when you create this PR, otherwise reviewers will be assigned for you.

If you have made additions or changes to the documentation, or if you _need_ documentation for these code changes, please make sure a technical writer has looked it over.

Once the reviewer(s) accept the PR and the branch passes continuous integration, the PR is eligible for merge.

At that time, if you have commit access (are an OpenNMS Group employee or a member of the Order of the Green Polo) you are welcome to merge the PR.
Otherwise, a reviewer can merge it for you.

Thanks for taking time to contribute!

### External References

* JIRA (Issue Tracker): http://issues.opennms.org/browse/NMS-13040

### CODE REVIEW

**Comment by cgorantla** [comment]
Provision UI before this change 
Screenshot from 2020-11-17 11-29-25


Provision UI after this change 
Screenshot from 2020-12-04 14-09-32

**Review by RangerRick — APPROVED** [review]

**Comment by indigo423** [comment]
Definitely better than the old one, here a suggestion with radio buttons instead of the explanation of the buttons with Full/Partial/DB-Only/True/False/dbonly + explanation.

Screenshot 2020-12-07 at 16 22 29

**Comment by cgorantla** [comment]
> Definitely better than the old one, here a suggestion with radio buttons instead of the explanation of the buttons with Full/Partial/DB-Only/True/False/dbonly + explanation.
> 
> Screenshot 2020-12-07 at 16 22 29
> 
> It's not a blocker for me just an idea :)
> 
> https://lucid.app/lucidchart/invitations/accept/2fd883f8-ace1-47ab-addc-3580b439da0d

I see, We can move to radio buttons as it is saving us from defining tags like Full, Partial / DB Only which may or may not describe the behavior completely.

**Comment by Bonrob2** [comment]
Works for me!

**Comment by indigo423** [comment]
> Works for me!

@Bonrob2 @cgorantla I've seen the PR was targeted against 28 which gives as a bit more freedom for changes :) I think the wording has to be revised. We have
* Policies
* Detectors
* SNMP scan
Is DB-Only skipping all the above or just the "SNMP scan"?

**Comment by cgorantla** [comment]
> > Works for me!
> 
> @Bonrob2 @cgorantla I've seen the PR was targeted against 28 which gives as a bit more freedom for changes :) I think the wording has to be revised. We have
> 
> * Policies
> * Detectors
> * SNMP scan
> 
> Is DB-Only skipping all the above or just the "SNMP scan"?

I would suggest to use the wording in this PR.
DB Only doesn't scan anything. It synchronize any updates in the requisition like categories or node label changes to DB. 
Scan here refers to all of the above ( Policies/Detectors/SNMP Scan).  
I see no reason to refer to SNMP scan separately.

**Comment by Bonrob2** [comment]
> 
> I would suggest to use the wording in this PR.
> DB Only doesn't scan anything. It synchronize any updates in the requisition like categories or node label changes to DB.
> Scan here refers to all of the above ( Policies/Detectors/SNMP Scan).
> I see no reason to refer to SNMP scan separately.

So the last radio button could be "Synchronize all nodes, skip scanning phase"?

**Comment by indigo423** [comment]
> > > Works for me!
> > 
> > 
> > @Bonrob2 @cgorantla I've seen the PR was targeted against 28 which gives as a bit more freedom for changes :) I think the wording has to be revised. We have
> > 
> > * Policies
> > * Detectors
> > * SNMP scan
> > 
> > Is DB-Only skipping all the above or just the "SNMP scan"?
> 
> I would suggest to use the wording in this PR.
> DB Only doesn't scan anything. It synchronize any updates in the requisition like categories or node label changes to DB.
> Scan here refers to all of the above ( Policies/Detectors/SNMP Scan).
> I see no reason to refer to SNMP scan separately.

Totally agree with you. "Just import without scanning" would be fine. What do you think about this one:

Screenshot 2020-12-07 at 18 10 49

**Comment by cgorantla** [comment]
> > > > Works for me!
> > > 
> > > 
> > > @Bonrob2 @cgorantla I've seen the PR was targeted against 28 which gives as a bit more freedom for changes :) I think the wording has to be revised. We have
> > > 
> > > * Policies
> > > * Detectors
> > > * SNMP scan
> > > 
> > > Is DB-Only skipping all the above or just the "SNMP scan"?
> > 
> > 
> > I would suggest to use the wording in this PR.
> > DB Only doesn't scan anything. It synchronize any updates in the requisition like categories or node label changes to DB.
> > Scan here refers to all of the above ( Policies/Detectors/SNMP Scan).
> > I see no reason to refer to SNMP scan separately.
> 
> Totally agree with you. "Just import without scanning" would be fine. What do you think about this one:
> 
> Screenshot 2020-12-07 at 18 12 55

Perfect. Wording looks good to me.

**Comment by Bonrob2** [comment]
I think the introductory text is too wordy, but understand that you want to explain what a scan is. Updated to make the language clearer:

`Choose a scan option to run detectors and policies, and fetch node and interface attributes from SNMP agents:`

**Comment by indigo423** [comment]
> I think the introductory text is too wordy, but understand that you want to explain what a scan is. Updated to make the language clearer:

Screenshot 2020-12-09 at 09 38 37

Updated the draft 💯

**Comment by RangerRick** [comment]
How dos this sound?

| Synchronize Requisition req1 |

Synchronize the requisition, applying any changes to the database and fetching node and interface attributes from SNMP. Additionally:

[x] Full scan of all nodes
[ ] Full scan of only new nodes
[ ] No scan, just synchronize

  [ Synchronize ] [ Cancel ]

**Comment by RangerRick** [comment]
I'm not sold on "Additionally:" but haven't thought of a better way to say "and also..."

**Comment by pbrane** [comment]
The Synchronize is implied.  I would change the 3 options to simply make the choice related to the scan option:

Following synchronization:
    [x] Scan all nodes
    [ ] Scan added nodes only
    [ ] No scanning

or what @RangerRick says. ;)

**Comment by Bonrob2** [comment]
Could we

**Comment by Bonrob2** [comment]
Sorry, didn't mean to close the PR!

Given David's comment, can we just remove the first sentence "Synchronize the requisition, applying any changes to the database and fetching node and interface attributes from SNMP."? 

It's a small prompt to have that much text on it. 

So it would just be:

| Synchronize Requisition req1 |

Following synchronization:
[x] Scan all nodes
[ ] Scan added nodes only
[ ] No scanning

[ Synchronize ] [ Cancel ]

**Comment by dino2gnt** [comment]
I think "scan only new nodes" or "scan only newly added nodes" makes the intent clearer, but other than that nit I like David's suggestion.

**Comment by cgorantla** [comment]
Updated UI with radio buttons. 
Screenshot from 2020-12-14 14-53-54

@RangerRick @Bonrob2 @indigo423  Let me know if that looks fine.

**Comment by Bonrob2** [comment]
For consistency, the OK should go on the left and Cancel on the right. 
We had discussed using "Synchronize" instead of "OK", but I'm fine with either.

**Comment by Bonrob2** [comment]
LGTM!

**Comment by cgorantla** [comment]
Updated.

Screenshot from 2020-12-15 10-45-07

**Comment by Bonrob2** [comment]
Looks great! Thanks for updating.

**Comment by indigo423** [comment]
Thank you very much everyone for the input! I have tears in my eyes - tears of joy :)

**Review by indigo423 — APPROVED** [review]
👍

**Comment by RangerRick** [comment]
Shia LeBouf - JUST DO IT.

**Comment by Bonrob2** [comment]
@cgorantla Is it Ok for me to merge this?

**Comment by cgorantla** [comment]
Upgraded `bootbox` to latest release version 5.5.2 since radio buttons are only supported in >5.0.
Verified screens where `bootbox` is getting used and didn't see any issues.
